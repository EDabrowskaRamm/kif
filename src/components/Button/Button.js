import React from 'react'

import { ButtonRoot, CloseButtonRoot } from './Button.shards'

const Button = ({ active = true, text, handleClick }) => {
  return (
    <ButtonRoot
      type='button'
      onClick={handleClick}
      disabled={!active}
    >
      {text}
    </ButtonRoot>
  )
}

const CloseButton = ({ handleClick }) => (
  <CloseButtonRoot
    type='button'
    onClick={handleClick}
  >
    &times;
  </CloseButtonRoot>
)

export { Button, CloseButton }