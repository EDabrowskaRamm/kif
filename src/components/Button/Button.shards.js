import styled from '@emotion/styled'

export const ButtonRoot = styled.button`
  border-radius: ${({ theme }) => theme.commons.borderRadius.button};
  background-color: ${({ theme }) => theme.colors.buttonActive.background};
  color: ${({ theme }) => theme.colors.buttonActive.text};
  padding: 16px 36px;
  outline: none;
  cursor: pointer;
  font-style: normal;
  font-weight: bold;
  font-size: 1.4rem;
  line-height: 1.6rem;

  &:disabled {
    color: ${({ theme }) => theme.colors.buttonDisabled.text};
    background: ${({ theme }) => theme.colors.buttonDisabled.background};
    cursor: default;
  }
`

export const CloseButtonRoot = styled.button`
  position: absolute;
  top: 20px;
  right: 20px;
  border-radius: ${({ theme }) => theme.commons.borderRadius.input};
  border: ${({ theme }) => theme.commons.borders.input};
  background: transparent;
  color: ${({ theme }) => theme.colors.text.title};
  padding: 10.5px 16.6px;
  font-size: 2.2rem;
  outline: none;
  cursor: pointer;
`