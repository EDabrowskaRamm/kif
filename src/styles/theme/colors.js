const _white = '#FFF'
const _colorGrey47 = '#707181'

const _tuna = '#30313D'
const _stormGray = _colorGrey47
const _snuff = '#DBDCEE'
const _luckyPoint = '#211965'
const _bittersweet = '#FF7165'

export const background = {
  primary: _white,
  overlay: _tuna,
}

export const text = {
  primary: _stormGray,
  secondary: _white,
  title: _tuna,
  error: _bittersweet,
}

export const border = {
  input: _snuff,
  error: _bittersweet,
}

export const buttonActive = {
  text: _white,
  background: _bittersweet
}

export const buttonDisabled = {
  text: _luckyPoint,
  background: 'rgba(141, 142, 161, 0.1)'
}