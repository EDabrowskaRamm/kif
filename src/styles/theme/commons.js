import { border } from './colors'

export const borders = {
  input: `1px solid ${border.input}`,
  error: `1px solid ${border.error}`,
}

export const borderRadius = {
  button: '4px',
  input: '10px',
  modal: '16px'
}