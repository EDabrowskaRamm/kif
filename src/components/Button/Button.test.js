import React from 'react'
import renderer from 'react-test-renderer'
import serializer from 'jest-emotion'
import { ThemeProvider } from 'emotion-theming'

import { Button, CloseButton } from './Button'

import theme from '../../styles/theme/theme'

expect.addSnapshotSerializer(serializer)

describe('renders buttons', () => {
  it('Snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <Button text='button' handleClick={() => { }} />
      </ThemeProvider>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Snapshot2', () => {
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <CloseButton handleClick={() => { }} />
      </ThemeProvider>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
