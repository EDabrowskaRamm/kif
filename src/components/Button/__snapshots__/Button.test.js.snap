// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders buttons Snapshot 1`] = `
.emotion-0 {
  border-radius: 4px;
  background-color: #FF7165;
  color: #FFF;
  padding: 16px 36px;
  outline: none;
  cursor: pointer;
  font-style: normal;
  font-weight: bold;
  font-size: 1.4rem;
  line-height: 1.6rem;
}

.emotion-0:disabled {
  color: #211965;
  background: rgba(141,142,161,0.1);
  cursor: default;
}

@media (min-width:992px) {
  .emotion-0 {
    font-size: 1.6rem;
  }
}

<button
  className="emotion-0"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  button
</button>
`;

exports[`renders buttons Snapshot2 1`] = `
.emotion-0 {
  position: absolute;
  top: 20px;
  right: 20px;
  border-radius: 10px;
  border: 1px solid #DBDCEE;
  background: transparent;
  color: #30313D;
  padding: 10.5px 16.6px;
  font-size: 2.2rem;
  outline: none;
  cursor: pointer;
}

<button
  className="emotion-0"
  onClick={[Function]}
  type="button"
>
  Ã—
</button>
`;
